noir-projects/aztec-nr/aztec/src/event/event_interface.nr

```diff
@@ -14,50 +14,136 @@ use dep::protocol_types::{
+/// | system) needs to make decisions based on the message. For example, consider a contract that escrows a  |
+/// | privately-stored NFT (i.e. an NFT represented by a note) and releases it to a buyer only after         |
+/// | receiving a payment in a specific token. Without on-chain delivery and guaranteed discoverability      |
+/// | afforded by constrained tagging and encryption, the buyer could potentially obtain the NFT without     |
+/// | sending a valid payment message (i.e. the token note hash preimage) to the seller, rugging the seller. |
+/// |                                                                                                        |
+/// | (To clarify the above, while the malicious buyer's payment token would still be deducted from their    |
+/// | balance, they would obtain the NFT while the seller would be unable to spend the payment token,        |
+/// | keeping the payment token note in limbo.)                                                              |
+/// +--------------------------------------------------------------------------------------------------------+
+///
+/// +--------------------------------------------------------------------------------------------------------+
+/// | 2. Unconstrained encryption, offchain message                                                          |
+/// |                                                                                                        |
+/// | This combination is useful when we either trust the sender to behave or the sender is incentivized to  |
+/// | deliver the message. To give you an example let's consider a payment app. When you are buying          |
+/// | groceries, the merchant can consider the payment successful once he receives the payment message, can  |
+/// | successfully decrypt it and see that its contents correspond to a valid token note.                    |
+/// +--------------------------------------------------------------------------------------------------------+
+///
+///
+/// Other combinations like unconstrained encryption, unconstrained tagging, private log stream don't seem
+/// to be meaningful because they are costlier than combination 2 above while providing no meaningful
+/// additional guarantees.
+pub struct MessageDeliveryEnum {
+    /// Constrained encryption + constrained tagging + private log stream
     ///
-    /// Only use in scenarios where the recipient not receiving the message is an acceptable outcome (e.g. because the
-    /// sender is somehow motivated to ensure the recipient learns of it).
-    pub NO_CONSTRAINTS: u8,
-    /// The contents of the log and its encryption are constrained. The tag (and therefore whether the recipient is
-    /// actually able to find the message) is not.
+    /// This gives us an on-chain guarantee that a recipient is able to discover and receive a message. See
+    /// combination 1 above for details.
+    pub CONSTRAINED_ONCHAIN: u8,
```

Comment:
This is not incorrect, but it's a lie - we do not support this at all. We don't have a way to do constrained tagging. It is being designed and still under discussion. Sure, that's what we want to convey I guess. It's just not true right now

Reply:
Oooh an opportunity to shill one of my hackmds! Jan, check this out: https://hackmd.io/MY7QqIaxTg-a2CMu_Znu4Q
You'll love it. Better than Harry Potter.